### PR TITLE
feat(records): grey ou part time days

### LINKED ISSUE #12: Dimmed part time days
**Context**

Some users will have a part-time contract, with a default part-time day. It would be nice if we have a indicator for this.

We also want to check if we can link this with the Intracto application, otherwise we fill it ourselves and store it in the firestore.

**Goal**

Get users part-time day from either firestore of the Intracto application and use this data to grey out the fields like we do with the holidays.

### PR DESCRIPTION
**Pull request by italoteix**
# Changes

## Related issues

Resolves #12 

## Added

- Add `isPartTime` to `WeekDay` type
- Part-time workdays have a slightly darker background

## Removed

N/A

## Changed

N/A

## How to test

- If you don't have a part-time work-scheme you can either ask to temporarily change it on Bridge to one of the admins or mockup the response from the work-schema-service
- Check if the background for part-time days have a different color

## Screenshots

Screenshot 2021-07-23 at 09 03 34

### CODE REVIEW

**Comment by vladPinteaFrontmen** [comment]
I feel like more explanation will be needed, at the employee's end, as to why it's grayed out, so a label like we do for leave, holiday, weekend makes sense to me, or a title on hover.

**Comment by lassecapel** [comment]
ideally, you should also see half days, some people have a half part time day. For example 4 hours on a friday

File: helpers/dates.ts [hunk 1]
```diff
@@ -5,136 +5,141 @@ import {
...
-  workScheme?: WorkScheme[]
+  workScheme?: WorkScheme[],
 ) {
-  return days.map((day) => {
-    const isCustomHoliday = customHolidays.includes(day.date);
+  return days.map(day => {
+    const isCustomHoliday = customHolidays.includes(day.date)
+
+    const workSchemeDay = workScheme?.find(ws => ws.date === day.date)
+    const isPublicHoliday = !!workSchemeDay?.holiday
```
Comment (sanderdejong88) [inline helpers/dates.ts:73]: Why are we casting it to a boolean here? I would try to keep those kind of conversions as close to the api call as possible. That way you don't keep repeating yourself.

**Comment by italoteix** [comment]
Added caption for part-time days
Screenshot 2021-07-27 at 12 55 09

File: helpers/dates.ts (hunk 1 above)
Comment (italoteix) [reply helpers/dates.ts]: Even converting it to a boolean, this check is still needed as `workScheme` and `workSchemeDay` can be `undefined` at this point

File: components/records/weekly-timesheet-row.vue [hunk 2]
```diff
@@ -72,20 +72,21 @@ import {
   defineComponent,
   PropType,
   watch,
-} from "@nuxtjs/composition-api";
```
Comment (vladPinteaFrontmen) [inline components/records/weekly-timesheet-row.vue:75]: not sure if it was you or something from the linter but it seems all semicolons have been removed from this file

**Review by vladPinteaFrontmen — APPROVED** [review]
short of that file missing semicolons, code looks ok. I assume you are still waiting on @sanderdejong88 's input but otherwise it's good.

**Pull request merged**